src/main/java/org/skypro/skyshop/service/BasketService.java

```diff
@@ -0,0 +1,38 @@
+package org.skypro.skyshop.service;
+
+import org.apache.catalina.User;
+import org.skypro.skyshop.model.basket.BasketItem;
+import org.skypro.skyshop.model.basket.ProductBasket;
+import org.skypro.skyshop.model.basket.UserBasket;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Service
+public class BasketService {
+    private final ProductBasket basket;
+    private final StorageService storageService;
+
+    public BasketService(ProductBasket basket, StorageService storageService) {
+        this.basket = basket;
+        this.storageService = storageService;
+    }
+
+    public void add(UUID id) {
+        if (storageService.getProductById(id).isPresent() == false) {
```

Comment:
Используй инверсию условия

```suggestion
        if (!storageService.getProductById(id).isPresent()) {
```
или метод isEmpty

```suggestion
        if (storageService.getProductById(id).isEmpty()) {
```

Reply:
Обычно я использую isEmpty, но в задании было сказано именно так: isPresent == false